### ISSUE TITLE
Catch errors which result because of PHP's auto conversion of numeric keys to integers

### ISSUE THREAD

**Issue by ostrolucky** [vimeo/psalm#1658]
https://psalm.dev/r/9fa42439ec This doesn’t catch real error and gives bogus one instead https://3v4l.org/pu96D

```php
<?php

/**
 * @param string[] $arr
 * @return array<string, null>
 */
function foo(array $arr) : array {
    return array_fill_keys($arr, null);
}
```

**Comment by muglug** [comment]
Do you have a suggestion as to how they could be caught?

This is basically just some fundamentally unsound PHP behaviour (that Hack, as part of its divergence from PHP, intends to imminently).

**Comment by ostrolucky** [comment]
True, psalm can't know input can be numeric strings. Is it possible to specify it? To have `numeric[]` or similar, instead of just `string[]`?

**Comment by muglug** [comment]
Basically not possible, I'm afraid.
